exercises/practice/dnd-character/dnd_character_test.rb

```diff
@@ -0,0 +1,56 @@
+require 'minitest/autorun'
+require_relative 'dnd_character'
+
+module Helpers
+  BASE_HITPOINTS = 10
+  CONSTITUTION_MODIFIER_TO_CONSTITUTION_MAP = {
+    -4 => [3],
+    -3 => [4, 5],
+    -2 => [6, 7],
+    -1 => [8, 9],
+    0 => [10, 11],
+    1 => [12, 13],
+    2 => [14, 15],
+    3 => [16, 17],
+    4 => [18]
+  }.freeze
+
+  def attributes
+    %i[strength dexterity constitution intelligence wisdom charisma]
+  end
+end
+
+class DndCharacterTest < Minitest::Test
+  include Helpers
+
+  10.times do |hitpoint_check|
+    define_method "test_hipoints_correctly_calculated_test #{hitpoint_check}" do
+      character = DndCharacter.new
+      constitution_modifier = character.hitpoints - BASE_HITPOINTS
+
+      assert_includes CONSTITUTION_MODIFIER_TO_CONSTITUTION_MAP[constitution_modifier],
+        character.constitution,
+        "Hitpoints: #{character.hitpoints}, Constitution: #{character.constitution}"
+    end
+  end
+
+  def test_random_character_attributes
+    character = DndCharacter.new
+    allowed_range = 3..18
+    expected_hitpoints = BASE_HITPOINTS + (character.constitution / 2 - 5)
+
+    attributes.each do |attribute|
+      assert_includes allowed_range, character.send(attribute)
+    end
+
+    assert_equal expected_hitpoints, character.hitpoints
+  end
+
+  def test_attributes_being_accessed_do_not_change
+    character = DndCharacter.new
+
+    (attributes << :hitpoints).each do |attribute|
+      assert_equal character.send(attribute), character.send(attribute)
+    end
```

Comment:
I do not have much confidence that this is doing well what the test says it is doing.

```ruby
    (attributes << :hitpoints).each do |attribute|
      expected = character.send(attribute)
      10.times do
        assert_equal expected, character.send(attribute)
      end
```
This would check each 10 (arbitrary choice) times, against what it was the answer for the first call